public/scripts/project/remote.js

```diff
@@ -0,0 +1,142 @@
+define(function(require) {
+  var $ = require("jquery");
+
+  var Path = Bramble.Filer.Path;
+  var Buffer = Bramble.Filer.Buffer;
+
+  // Installs a tarball (arraybuffer) containing the project's files/folders.
+  function installTarball(fs, root, tarball, callback) {
+    var untarWorker;
+    var pending = null;
+    var sh = new fs.Shell();
+
+    function extract(path, data, callback) {
+      path = Path.resolve(root, path);
+      var basedir = Path.dirname(path);
+
+      sh.mkdirp(basedir, function(err) {
+        if(err && err.code !== "EEXIST") {
+          return callback(err);
+        }
+
+        fs.writeFile(path, new Buffer(data), {encoding: null}, callback);
+      });
+    }
+
+    function finish(err) {
+      untarWorker.terminate();
+      untarWorker = null;
+
+      callback(err);
+    }
+
+    function writeCallback(err) {
+      if(err) {
+        console.error("[Thimble error] couldn't extract file for tar", err);
+      }
+
+      pending--;
+      if(pending === 0) {
+        finish(err);
+      }
+    }
+
+    untarWorker = new Worker("/scripts/editor/vendor/bitjs-untar-worker.min.js");
+    untarWorker.addEventListener("message", function(e) {
+      var data = e.data;
+
+      if(data.type === "progress" && pending === null) {
+        // Set the total number of files we need to deal with so we know when we're done
+        pending = data.totalFilesInArchive;
+      } else if(data.type === "extract") {
+        extract(data.unarchivedFile.filename, data.unarchivedFile.fileData, writeCallback);
+      } else if(data.type === "error") {
+        finish(new Error("[Thimble error]: " + data.msg));
+      }
+    });
+
+    untarWorker.postMessage({file: tarball});
+  }
+
+  function loadTarball(fs, root, host, callback) {
+    // jQuery doesn't seem to support getting the arraybuffer type
+    var url = host + "/getFileContents";
```

Comment:
Won't hurt to do it.  Does it require any changes to server side to strip that at all?